graftm/decorator.py

```diff
@@ -0,0 +1,80 @@
+
+# System imports
+import logging
+from skbio import TreeNode
+from graftm.reannotator import Reannotator
+from graftm.tree_decorator import TreeDecorator
+
+class Decorator:
+    '''re-root a tree and decorate it using old taxonomy, for the graftM 
+    decorate pipeline.
+    
+    ALL trees provided to this class must be in newick format. No other formats 
+    are supported. If any taxonomic decoration is already in the tree to be 
+    decorated by this class is WILL be overwritten. Of course, bootstrap values 
+    will remain untouched.'''
+
+    def __init__(self, **kwargs):
+        '''
+        Parameters
+        ----------
+        reference_tree_path: str
+            Path to the file containing the reference tree, which is used to
+            retroot the tree tree provided to old_tree
+        old_tree_path: str
+            Path to the file containing the tree to be re-rooted. This tree will
+            be rerooted at the same position as the tree porovided to the 
+            reference_tree
+        '''
+        reference_tree_path = kwargs.pop('reference_tree_path', None)
+        tree_path = kwargs.pop('tree_path', None)
```

Comment:
Well in the newer classes we've been not having an init, and instead the 'main' function has kwargs, and there's no instance variables. I always struggle to know what should be a parameter and what should be an instance variables, and tend towards parameters. All the details are then provided in one place, then - easier to read, and more self-contained.

So it should be easy, do you mind doing the same here, just giving the reference tree and old tree to a `decorate` method (perhaps more descriptive than `main`)?